lambdaclass/mirra_backend
# Add modifier operation to message

## Motivation

The client needs a way to be able to tell if a modifier is positive or negative. The simplest way is to let it know which operation is being performed with the value.
Closes https://github.com/lambdaclass/champions_of_mirra/issues/418

## Summary of changes

- Rename some Modifier fields that were unnecesarily long
- Change the protobuf message to include the operation

## How to test it?

1. Move `apps/gateway/test/support/socket_tester.ex` under `apps/gateway/lib/gateway`
2. Configure inspect limit 
```
IEx.configure(inspect: [limit: :infinity])
```
3. Create a user:
```
{:ok, user} = Champions.Users.register("Lionel Messi")
```
4. Fetch the first level of the campaign
```
level_id = user.super_campaign_progresses |> hd() |> Map.get(:level_id)
```
5. Create a SocketTester instance
```
{:ok, pid} = Gateway.SocketTester.start_link()
```
6. Battle a level to check the response
```
Gateway.SocketTester.fight_level(pid, user.id, level_id)
```
```
send(pid, {:last_message, self()})
flush()
```

You should be able to see the operation in any modifier actions:
```
           %Gateway.Serialization.Action{
             action_type: {:modifier_received,
              %Gateway.Serialization.ModifierReceived{
                skill_id: "c16283d6-dc37-468f-9dcb-085ad82e378f",
                target_id: "96a09192-b594-48e8-bc21-2f38161f6100",
                stat_affected: %Gateway.Serialization.StatAffected{
                  stat: :ATTACK,
                  amount: 1.0800000429153442,
                  __unknown_fields__: []
                },
                operation: "Multiply",
                __unknown_fields__: []
              }},
             __unknown_fields__: []
           }
```
